device/sycl/src/seeding/seed_selecting.sycl

```diff
@@ -0,0 +1,313 @@
+            auto& spB = internal_sp_device.bin(spB_loc.bin_idx)[spB_loc.sp_idx];
+            auto& spT = internal_sp_device.bin(spT_loc.bin_idx)[spT_loc.sp_idx];
+
+            // if the number of seeds reaches the threshold, break
+            if (n_seeds_per_spM >= m_filter_config.maxSeedsPerSpM + 1) {
+                break;
+            }
+
+            seed aSeed({spB.m_link, spM.m_link, spT.m_link, aTriplet.weight,
+                        aTriplet.z_vertex});
+
+            // check if it is a good triplet
+            if (seed_selecting_helper::cut_per_middle_sp(
+                    m_filter_config, spacepoints_device, aSeed,
+                    aTriplet.weight) ||
+                n_seeds_per_spM == 0) {
+
+                vecmem::atomic<uint32_t> obj(&num_seeds);
+                auto pos = obj.fetch_add(1);
+
+                // prevent overflow
+                if (pos >= seeds.size()) {
+                    break;
+                }
+                n_seeds_per_spM++;
+
+                seeds[pos] = aSeed;
+            }
+        }
+    }
+
+    private:
+    const seedfilter_config m_filter_config;
+    spacepoint_container_view m_spacepoints_view;
+    sp_grid_view m_internal_sp_view;
+    doublet_counter_container_view m_doublet_counter_view;
+    triplet_counter_container_view m_triplet_counter_view;
+    triplet_container_view m_triplet_view;
+    seed_container_view m_seed_view;
+    local_accessor<triplet> m_localMem;
+};
+
+void seed_selecting(const seedfilter_config& filter_config,
+                    host_spacepoint_container& spacepoints,
+                    sp_grid& internal_sp,
+                    host_doublet_counter_container& doublet_counter_container,
+                    host_triplet_counter_container& triplet_counter_container,
+                    host_triplet_container& triplet_container,
+                    host_seed_container& seed_container,
+                    vecmem::memory_resource& resource, ::sycl::queue* q) {
+
+    auto spacepoints_view = get_data(spacepoints, &resource);
+    auto doublet_counter_container_view =
+        get_data(doublet_counter_container, &resource);
+    auto triplet_counter_container_view =
+        get_data(triplet_counter_container, &resource);
+    auto triplet_container_view = get_data(triplet_container, &resource);
+    auto seed_container_view = get_data(seed_container, &resource);
+    auto internal_sp_view = get_data(internal_sp, resource);
+
+    // The thread-block is desinged to make each thread find triplets per
+    // compatible middle-bot doublet
+
+    // -- localSize
+    // The dimension of workGroup (block) is the integer multiple of WARP_SIZE
+    // (=32)
+    unsigned int localSize = 64;
+    // Calculate the global number of threads to run in kernel
+    unsigned int num_groups = 0;
+    for (unsigned int i = 0; i < internal_sp.nbins(); ++i) {
+        num_groups +=
+            triplet_counter_container.get_headers()[i].n_mid_bot / localSize +
```

Comment:
@konradkusiak97 you may want to check #119 to fix a notorious bug that I had... :)

Reply:
Right, I will fix that... thanks! :wink: